salt/output/__init__.py

```diff
@@ -125,7 +125,12 @@ def display_output(data, out=None, opts=None, **kwargs):
                     ofh.close()
             return
         if display_data:
-            salt.utils.stringutils.print_cli(display_data)
+            if six.PY2:
+                salt.utils.stringutils.print_cli(display_data)
```

Comment:
Would it make more sense to just fix the underlying `print_cli`?

I believe that it would be as simple as changing `print(msg.encode('utf-8'))` to `sys.stdout.buffer.write(msg)`. In either case, I'd love to see some tests for this - it should be pretty trivial to add something in the unit tests dir. Then I'll be a hearty 👍 on this.

Reply:
@saltstack/team-core any thoughts on whether this is the right place for the fix, or we should fix the underlying `stringutils.print_cli`?

Reply:
I changed it here in order to be sure future improvements to `print_cli` won't affect the information transport, because `print_cli` sounds like something relevant for interactive stuff. Also, I didn't want to break other uses of the function unintentionally.

I'm of course open to suggestion for a better fix :)

Reply:
@saltstack/team-core any thoughts?